<launch>
  <arg name="tello_ip" default="192.168.10.1" />
  <arg name="tello_cmd_server_port" default="8889" />
  <arg name="local_cmd_client_port" default="8890" />
  <arg name="local_vid_server_port" default="6038" />
  <!-- TODO(Jerome) -->
  <!-- <arg name="camera_calibration"    default="$(find tello_driver)/cfg/960x720.yaml" /> -->
  <arg name="namespace" default="tello" />
  <arg name="vel_cmd_scale" default="1.0" />
  <arg name="connect_timeout_sec" default="10.0" />
  <!-- percent -->
  <arg name="low_bat_threshold" default="7" />
  <arg name="altitude_limit" default="10" />
  <arg name="attitude_limit" default="15" />
  <arg name="publish_raw_image" default="False" />

  <push-ros-namespace namespace="$(var namespace)"/>

  <node pkg="tello_driver" name="tello_driver_node" exec="tello_driver_node" output="screen" args="--ros-args --log-level info">
    <param name="local_cmd_client_port" value="$(var local_cmd_client_port)" />
    <param name="local_vid_server_port" value="$(var local_vid_server_port)" />
    <param name="tello_ip" value="$(var tello_ip)" />
    <param name="tello_cmd_server_port" value="$(var tello_cmd_server_port)" />
    <param name="connect_timeout_sec" value="$(var connect_timeout_sec)" />
    <param name="stream_h264_video" value="false" />
    <!-- <param name="camera_calibration" value="$(var camera_calibration)" /> -->
    <param name="odom_frame" value="odom"/>
    <param name="tf_prefix" value=""/>
    <param name="vel_cmd_scale" value="$(var vel_cmd_scale)"/>
    <param name="low_bat_threshold" value="$(var low_bat_threshold)"/>
    <param name="altitude_limit" value="$(var altitude_limit)"/>
    <param name="attitude_limit" value="$(var attitude_limit)"/>
  </node>

    <!-- <node if="$(var publish_raw_image)" pkg="image_transport" name="decompress_image" type="republish" args="h264 in:=image_raw raw out:=image_raw" /> -->

</launch>
